Decide/src/main/java/DD2480/decide/CMVHandler.java

```diff
@@ -83,14 +81,34 @@ private boolean licTwo() {
 	}
 
 	private boolean licThree() {
-		// TODO Auto-generated method stub
-		return false;
+        for (int index = 0; index < dataPoints.length - 2; index++) {
+            if (GeometryHelper.area(dataPoints[index], dataPoints[index + 1], dataPoints[index + 2]) > parameters.area1)
+                return true;
+        }
+        return false;
 	}
 
 	private boolean licFour() {
-		// TODO Auto-generated method stub
-		return false;
-	}
+        for (int index = 0; index < (dataPoints.length - parameters.qPts + 1); index++) {
```

Comment:
perhaps you should start check if quad is valid and that there is enough numpoints:

(2  =< Q PTS =< NUMPOINTS), (1  <= QUADS  <= 3)